### PR TITLE
Add tests for Agent

### PR DESCRIPTION
**Pull request by bobisme**
### Background

`agent.py` is barely covered by tests. This is my beginning take at adding tests, starting slow with smaller functions and doing refactoring to get familiar with the module before taking on the massive `start_interaction_loop`.

### Changes

* Added tests for `Agent._resolve_pathlike_command_args`
* Refactored `Agent.get_self_feedback_loop` to use formatted strings to hopefully be a little clearer.
* Added a test for `Agent.get_self_feedback_loop` which checks the format for specific information.
* Removed duplicate tests in `tests/unit/test_get_self_feedback.py`

### Documentation

Added tests. Behavior and documentation unchanged.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: tests/test_agent.py [hunk 1]
```diff
@@ -34,14 +34,50 @@ def agent():
 
 def test_agent_initialization(agent):
     assert agent.ai_name == "Test AI"
-    assert agent.memory == agent.memory
```
Comment (bobisme) [inline tests/test_agent.py]: removed these because they test nothing. like `assert 1 == 1`

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Boostrix** [comment]
this is something that looks useful and compact in its current form, so could probably be considered as a candidate to be reviewed/integrated despite the ongoing re-arch effort - especially since, testing is needed for the re-arch as well

**Comment by Pwuts** [comment]
Conflicts are resolved, but one test fails. @bobisme would you mind fixing the error so we can continue with review/merge?

**Comment by Pwuts** [comment]
The functionality corresponding to these tests has been removed, so I'll close this. Still, thanks for submitting! If you ever feel like contributing again, let us know on Discord. :)

**Pull request closed without merging**